src/Generator/JsonGenerator.php

```diff
@@ -91,6 +91,16 @@ private function generateToc(MetaEntry $metaEntry, ?array $titles): array
 
     private function guessNext(string $parserFilename): ?array
     {
+        $meta       = $this->getMeta($parserFilename);
+        $parentFile = $meta->getParent();
+
+        if ('index' === $parentFile && \count($tocs = $meta->getTocs()) === 1 && \count($tocs[0]) > 0) {
```

Comment:
Can you explain this with a comment? This is the type of stuff that I definitely want a test covering - to help us know why this is here and not accidentally break it :)

Reply:
longer-term, I'd like to think about some other way to do this. Like maybe these "index" pages... that are not *really* part of the docs - are given some sort of option on top `:index-page:` - not sure... I just don't love enforcing this convention deep down here.